attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -2,25 +2,51 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 //
-
-use crate::utils::pad;
-
+use super::tsm_report::*;
 use super::Attester;
 use anyhow::*;
 use base64::Engine;
-use log::debug;
-use scroll::Pread;
 use serde::{Deserialize, Serialize};
 use std::path::Path;
-use tdx_attest_rs::{self, tdx_report_t};
 
 mod report;
 
+#[cfg(feature = "tdx-dcap-ioctl")]
+use tdx_attest_rs::{self, tdx_report_t};
+
 const TDX_REPORT_DATA_SIZE: usize = 64;
 const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
 
+fn tdx_getquote_ioctl_is_available() -> bool {
+    if cfg!(feature = "tdx-dcap-ioctl") {
+        Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()
```

Comment:
I have tried this PR with kernel 5.19, and it works well. but 
for kernel 6.x, the device path changed to `/dev/tdx_guest`, 
do you consider add this check for kernel 6.x?

Reply:
adding support for `/dev/tdx_guest` belongs to a separate PR. Let me prepare that